lib/wash/exe/wash.js

```diff
@@ -531,33 +531,53 @@ Wash.prototype.readEvalPrintLoop = function() {
 };
 
 /**
- * Parse a line of shell input into the path and the argument string.
- *
- * @return {Array<string>}
+ * @constructor
  */
-Wash.prototype.parseShellInput = function(str) {
-  var path, argv, redirMode, redirPath;
+Wash.RedirectInfo = function(operator, pathSpec) {
+  this.pathSpec = pathSpec;
+  this.mode = Wash.REDIR_OP_TO_MODE_[operator];
+  if (!this.pathSpec || !this.mode)
+    throw new AxiomError.Invalid('I/O redirection', operator + pathSpec);
+};
 
-  var pos = str.indexOf('>');
-  if (pos > -1) {
-    if (str[pos + 1] === '>') {
-      redirMode = 'wc';
-      redirPath = str.substr(pos + 2).trim();
-    } else {
-      redirMode = 'wct';
-      redirPath = str.substr(pos + 1).trim();
-    }
-    str = str.substr(0, pos);
-  }
+/**
+ * @constructor
+ */
+Wash.Command = function(pathSpec, argv, redirIn, redirOut) {
+  this.pathSpec = pathSpec;
+  this.argv = argv;
+  this.redirIn = redirIn;
+  this.redirOut = redirOut;
+};
 
-  pos = str.indexOf(' ');
-  if (pos > -1) {
-    path = str.substr(0, pos);
-    argv = str.substr(pos + 1).trim();
-  } else {
-    path = str;
-    argv = null;
-  }
+/**
+ * Matches a command line with possible stdout redirection to a file:
+ *
+ *    command [arg1 arg2 ...] [< some_file] [> some_file]
+ *
+ * @private
+ * @const
+ */
+Wash.CMD_WITH_IO_REDIRECT_RE_ = /^([^<>]+?)(?:\s+([^<>]+?))?\s*([<>].+)?$/;
+Wash.STDIN_REDIRECT_RE_ = /(<)\s*([^<>\s]+)?/;
+Wash.STDOUT_REDIRECT_RE_ = /(>>?)\s*([^<>\s]+)?/;
+Wash.REDIR_OP_TO_MODE_ = {
+  '<': 'r',
+  '>': 'wct',
+  '>>': 'wc'
+};
+
+/**
+ * Parse a line of shell input into the path and the argument string.
+ *
+ * @return {!Wash.Command}
+ */
+Wash.prototype.parseShellCommand = function(str) {
```

Comment:
These are just a quick stop-gap solution to get the thing going. This parsing is quick and dirty anyway, and there is an outstanding #200 to fix this ASAP. I will tackle it immediately after this.